Kernel/Heap/kmalloc.cpp

```diff
@@ -223,17 +278,36 @@ static inline void kmalloc_verify_nospinlock_held()
     }
 }
 
+static void* allocate_new_eternal_heap_slot()
+{
+    // Note: Only Eternal heaps should be in the static eternal heap of the kernel,
+    // therefore each chunk in the static eternal heap must of an EternalHeap object and only it!
+    kmalloc_verify_nospinlock_held();
+
+    size_t size = round_up_to_power_of_two(sizeof(EternalHeap), sizeof(void*));
```

Comment:
The static_assert earlier guarantees that this is equivalent to `size_t size = sizeof(EternalHeap);`